### PR TITLE
add e2e test for kubectl default container annotation

### PR DESCRIPTION
**Pull request by pacoxu**
#### What type of PR is this?
/kind cleanup
/kind feature

#### Which issue(s) this PR fixes:

Fixes https://github.com/kubernetes/enhancements/pull/3457

#### Does this PR introduce a user-facing change?

```release-note
kubectl: add e2e test for default container annotation
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs
- [KEP]: https://github.com/kubernetes/enhancements/blob/master/keps/sig-cli/2227-kubectl-default-container/README.md
```

### CODE REVIEW

File: test/e2e/kubectl/default_container.go [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+	"github.com/onsi/ginkgo/v2"
+	"k8s.io/kubernetes/test/e2e/framework"
+	admissionapi "k8s.io/pod-security-admission/api"
+)
+
+var _ = SIGDescribe("Kubectl default container", func() {
+	f := framework.NewDefaultFramework("default-container")
+	f.NamespacePodSecurityEnforceLevel = admissionapi.LevelBaseline
+
+	ginkgo.Describe("logs", func() {
```
Comment (soltysh) [inline test/e2e/kubectl/default_container.go:33]: While you're at it, can you move all the `kubectl logs` related tests over to this file, and rename it to `logs.go` this way we'll start splitting that gigantic https://github.com/kubernetes/kubernetes/blob/master/test/e2e/kubectl/kubectl.go into smaller pieces.

Comment (soltysh) [reply test/e2e/kubectl/default_container.go:33]: Can be done in a followup.

File: test/e2e/kubectl/logs.go [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+					Image: agnhostImage,
+					Args:  []string{"logs-generator", "--log-lines-total", "20", "--run-duration", "5s"},
+				},
+			},
+			RestartPolicy: v1.RestartPolicyNever,
+		},
+	}
+}
+
+// TODO promote to conformance test
```
Comment (soltysh) [inline test/e2e/kubectl/logs.go:72]: Nit: drop this, not needed.

**Review by soltysh — APPROVED** [review]
/lgtm
/approve
It would be nice to split this into 2 commits:
1. move existing test to new file
2. add the new e2e

**Comment by soltysh** [comment]
/assign @dims 
for conformance approval
Dims this is just moving the pre-existing conformance test about `kubect logs` from that one gigantic kubectl.go file into smaller, more manageable pieces, and adds a new test covering that area (not conformance test).

**Comment by dims** [comment]
/approve
/lgtm

thanks for the context @soltysh

**Pull request merged**